src/components/UniqueDomain/index.tsx

```diff
@@ -65,15 +65,27 @@ export const UniqueDomain: React.FC<{
           clearTimeout(timer)
 
           if (!validityReq.ok) {
-            const message = `Error validating domain: ${validityReq.statusText}`
-            console.error(message) // eslint-disable-line no-console
-            setError(message)
+            const responseBody = await validityReq.json()
+
+            let errorMessage = `Error validating domain: ${validityReq.statusText}`
```

Comment:
You might be able to simplify some of this by writing it like this:

```ts
  let errorMessage = responseBody?.error || `Error validating domain: ${validityReq.statusText}`
  if (responseBody.error === 'No subdomain provided.') {
    errorMessage = 'Please input a subdomain.'
  }
```

This will expose all API errors to the by default, while also allowing you to override them on a case-by-case basis as needed.